From akpm@osdl.org Mon May 15 21:52:17 2006
Message-Id: <200605160452.k4G4q8kK028913@shell0.pdx.osdl.net>
From: David Micon <DMicon@pelco.com>
Subject: USB: HID read busywait fix
To: DMicon@pelco.com, dtor_core@ameritech.net, greg@kroah.com, vojtech@suse.cz
Date: Mon, 15 May 2006 21:48:54 -0700


From: David Micon <DMicon@pelco.com>

Make a read of a HID device block until data is available.  Without it, the
read goes into a busy-wait loop until data is available.

Acked-by: Vojtech Pavlik <vojtech@suse.cz>
Cc: Dmitry Torokhov <dtor_core@ameritech.net>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/input/hiddev.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/usb/input/hiddev.c
+++ gregkh-2.6/drivers/usb/input/hiddev.c
@@ -317,6 +317,7 @@ static ssize_t hiddev_read(struct file *
 				}
 
 				schedule();
+				set_current_state(TASK_INTERRUPTIBLE);
 			}
 
 			set_current_state(TASK_RUNNING);
